From b5ac03001b66c409a37d8ed18379ffbd84e8323b Mon Sep 17 00:00:00 2001
From: Dimitri John Ledkov <dimitri.ledkov@surgut.co.uk>
Date: Tue, 25 Nov 2025 13:52:19 +0000
Subject: [PATCH] seed_src_jitter: always init jitter

Always init jitter entropy source, even when in error state. This
ensures that when primary DRBG is configured to use Jitter seed
source, jitter seed source remains in use.

Since, in the rare case that initialize fails, NULL is returned, and
the FIPS provider crng is initialised with NULL seed source, which
means default is used, which for now is OS-specific getrandom source.
---
 providers/implementations/rands/seed_src_jitter.c.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/providers/implementations/rands/seed_src_jitter.c.in b/providers/implementations/rands/seed_src_jitter.c.in
index 4d73f07574..3868bec48f 100644
--- a/providers/implementations/rands/seed_src_jitter.c.in
+++ b/providers/implementations/rands/seed_src_jitter.c.in
@@ -160,7 +160,7 @@ static int jitter_instantiate(void *vseed, unsigned int strength,
         ERR_raise_data(ERR_LIB_RAND, RAND_R_ERROR_RETRIEVING_ENTROPY,
                        "jent_entropy_init_ex (%d)", ret);
         s->state = EVP_RAND_STATE_ERROR;
-        return 0;
+        return 1;
     }
 
     s->state = EVP_RAND_STATE_READY;
-- 
2.51.0

